### ISSUE TITLE
Fix bug in creation of bounding-box ROIs

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#460]
The Cellpose task iterates over ROIs, which often means over _FOV_ ROIs. Say that we consider an MIP input, and then a FOV is a YxX array (e.g. 2160x2560). Let's ignore all the shape broadcasting (i.e. the fact that this is stored as a 1x1xYxX array on disk).

For each FOV ROI, the task runs Cellpose and computes the mask with the new labels, which is the YxX array `new_label_img`.

If we are also setting `output_ROI_table`, then for each FOV ROI the task also calls
```python
array_to_bounding_box_table(new_label_img, actual_res_pxl_sizes_zyx)
```
which currently looks like
```python
def array_to_bounding_box_table(
    mask_array: np.ndarray, pxl_sizes_zyx: list[float]
) -> pd.DataFrame:

    """
    Description

    :param dummy: this is just a placeholder
    :type dummy: int
    """

    labels = np.unique(mask_array)
    labels = labels[labels > 0]
    elem_list = []
    for label in labels:
        label_match = np.where(mask_array == label)
        # FIXME: multiplication of np.ndarray with list
        zmin, ymin, xmin = np.min(label_match, axis=1) * pxl_sizes_zyx
        zmax, ymax, xmax = (np.max(label_match, axis=1) + 1) * pxl_sizes_zyx

        length_x = xmax - xmin
        length_y = ymax - ymin
        length_z = zmax - zmin
        elem_list.append((xmin, ymin, zmin, length_x, length_y, length_z))

    df_columns = [
        "x_micrometer",
        "y_micrometer",
        "z_micrometer",
        "len_x_micrometer",
        "len_y_micrometer",
        "len_z_micrometer",
    ]

    if len(elem_list) == 0:
        df = pd.DataFrame(columns=[x for x in df_columns] + ["label"])
    else:
        df = pd.DataFrame(np.array(elem_list), columns=df_columns)
        df["label"] = labels

    return df
```

## Possible bug

Let's now consider a well which has two identical FOVs (as in "the image data are identical", for instance because we duplicated a FOV for testing purposes). The Cellpose segmentation will output the same array `new_label_img` twice, with the caveat that setting `relabeling=True` will shift the second-FOV mask values appropriately.

**What about the bounding boxes?**
A first test (to be confirmed) suggests that the list of bounding boxes will be wrong, since it will include the same set of label ROIs twice (once for the first FOV, and once for the second FOV). This is consistent with the code shown above, which never includes a spatial shift when creating the bounding boxes.

TL;DR The bounding-box-creation function produces ROIs that are defined relative to the FOV-ROI origin, but these are then combined together without the appropriate coordinate shifts.

## Why now?
I noticed this issue while testing two-steps labeling (first produce "organoid" labels and ROIs, and then run internal segmentation inside each ROI). Why did this issue never appear in the past, and especially as part of #306 examples (based on RS220304172545 data)? My first guess is that in that case primary labeling was performed at the whole-well level, so that there was a single primary-label ROI and this issue did not appear.

**Comment by tcompa** [comment]
Upon further checking, this is obviously a specific case of
* https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/339

**Comment by tcompa** [comment]
Ref fb3d6e43bee32f3a72f2d90f564d88b27b64567f and 7ec1d05e7a3bad477051c38aab950a2cd9f1db39 (part of #524).

These commits introduce a new argument `origin_zyx` (in pixels) for the `array_to_bounding_box_table` function, and provide the appropriate argument value when this is called from within the cellpose task. See
```python
def array_to_bounding_box_table(
    mask_array: np.ndarray,
    pxl_sizes_zyx: list[float],
    origin_zyx: tuple[int, int, int] = (0, 0, 0),
) -> pd.DataFrame:

    """
    TBD

    Args:
        mask_array: TBD
        pxl_sizes_zyx: TBD
        origin_zyx: If set, shift origin by this amount of pixels.
    """

    pxl_sizes_zyx_array = np.array(pxl_sizes_zyx)
    z_origin, y_origin, x_origin = origin_zyx[:]

    labels = np.unique(mask_array)
    labels = labels[labels > 0]
    elem_list = []
    for label in labels:

        # Compute bounding box
        label_match = np.where(mask_array == label)
        zmin, ymin, xmin = np.min(label_match, axis=1) * pxl_sizes_zyx_array
        zmax, ymax, xmax = (
            np.max(label_match, axis=1) + 1
        ) * pxl_sizes_zyx_array

        # Compute bounding-box edges
        length_x = xmax - xmin
        length_y = ymax - ymin
        length_z = zmax - zmin

        # Shift origin
        zmin += z_origin * pxl_sizes_zyx[0]
        ymin += y_origin * pxl_sizes_zyx[1]
        xmin += x_origin * pxl_sizes_zyx[2]

        elem_list.append((xmin, ymin, zmin, length_x, length_y, length_z))

....
```
and
```python
def cellpose_segmentation(...):
    ...

    logger.info(f"Now starting loop over {num_ROIs} ROIs")
    for i_ROI, indices in enumerate(list_indices):
        # Define region
        s_z, e_z, s_y, e_y, s_x, e_x = indices[:]
        region = (
            slice(s_z, e_z),
            slice(s_y, e_y),
            slice(s_x, e_x),
        )
        logger.info(f"Now processing ROI {i_ROI+1}/{num_ROIs}")

        ...

        if output_ROI_table:
            bbox_df = array_to_bounding_box_table(
                new_label_img,
                actual_res_pxl_sizes_zyx,
                origin_zyx=(s_z, s_y, s_x),
            )
```

**Comment by tcompa** [comment]
The solution described in the previous comment (and part of #524) now works as expected.
I set up a test with a well containing two FOVs. Then I run the cellpose task at the per-FOV level, and use a mocked segmentation version which produces the following "organoid" labels":
![Screenshot from 2023-09-18 13-00-53](https://github.com/fractal-analytics-platform/fractal-tasks-core/assets/3862206/b0f3c2bd-ea81-43eb-890d-43bc246f8674)

The corresponding bounding-box ROI table reads:
```python
    organoid_ROI_table.X: (
        array([[  0.  ,   0.  ,   0.  , 208.  , 175.5 ,   1.  ],
               [312.  , 263.25,   0.  , 104.  ,  87.75,   1.  ],
               [416.  ,   0.  ,   0.  , 208.  , 175.5 ,   1.  ],
               [728.  , 263.25,   0.  , 104.  ,  87.75,   1.  ]], dtype=float32)
    )
```
where it is already clear that the third/fourth ROIs are correctly shifted so that they belong to the FOV on the right.

To make sure that everything works, I now run a "nuclear" segmentation using `input_ROI_table="organoid_ROI_table"`, and the nuclei labels do appear within the organoid-bounding-box regions:
![Screenshot from 2023-09-18 13-00-58](https://github.com/fractal-analytics-platform/fractal-tasks-core/assets/3862206/0c4b96e3-b65b-49b6-b9b6-5e09092d23f8)
